### PR TITLE
update spatial map alpha

### LINKED ISSUE #282: plot_spatial_map alpha_map does not work
matplotlib 3.3.1 no longer supports using an array of alpha values, instead it wants RGBA values in the image. needs to be updated or revert to matplotlib 3.1.3

### PR DESCRIPTION
**Pull request by leoscholl**
to close #282 i've updated plot_spatial_map to convert the data into rgba and apply the alpha layer there.

### CODE REVIEW

File: aopy/visualization.py [hunk 1]
```diff
@@ -247,6 +248,22 @@ def plot_spatial_map(data_map, x, y, alpha_map=None, ax=None, cmap='bwr'):
```
Comment (mmadduri) [inline aopy/visualization.py:248]: could you include what clim and show_bad are?

File: aopy/visualization.py [hunk 2]
```diff
@@ -261,18 +278,24 @@ def plot_spatial_map(data_map, x, y, alpha_map=None, ax=None, cmap='bwr'):
...
+        norm = cm.colors.Normalize(*clim)
+        scalarMap = cm.ScalarMappable(norm=norm, cmap=cmap)
+        data_map = scalarMap.to_rgba(data_map)
+
+        # Apply the alpha map after scaling from 0 to 1
         alpha_range = np.nanmax(alpha_map) - np.nanmin(alpha_map)
         alpha_map = (alpha_map - np.nanmin(alpha_map)) / alpha_range
         alpha_map[np.isnan(alpha_map)] = 0
-
+        data_map[:,:,3] = alpha_map
```
Comment (mmadduri) [inline aopy/visualization.py:293]: is it always 3? would it be more robust to add a new axis or something?

**Comment by mmadduri** [comment]
@leoscholl added some minor comments and will approve after you respond!

File: aopy/visualization.py (hunk 2 above)
Comment (leoscholl) [reply aopy/visualization.py:298]: yeah it's always 3, rgb are 0-2 and 3 is alpha. i think this is ok

**Comment by leoscholl** [comment]
thanks for the comments. good catch on the docstring! one of those parameters didn't do anything.. should be fixed now

**Review by mmadduri — APPROVED** [review]
looks good, thanks!

**Pull request merged**